### ISSUE TITLE
Minetest segfaults if audio can't be opened.

### ISSUE THREAD

**Issue by jastevenson303** [minetest/minetest#7250]
##### Issue type
 - Bug report
 
##### Minetest version
```
Minetest 0.5.0-dev-cca92537 (Linux)
Using Irrlicht 1.8.4
BUILD_TYPE=Release
RUN_IN_PLACE=0
USE_GETTEXT=0
USE_SOUND=1
USE_CURL=1
USE_FREETYPE=1
USE_LUAJIT=1
STATIC_SHAREDIR="/usr/local/share/minetest"
```

##### OS / Hardware
OS: Debian GNU/Linux OR Windows

##### Summary

Minetest will not start, by default, if the audio device on Linux is already in use, or can't be found.  Instead, it should imply --nosound and print a warning that audio is disabled.

Actually, I just `minetest --help`'d and I don't see a --nosound anyway, so perhaps this is actually a feature request for that as well.

##### Steps to reproduce

```
Cannot connect to server socket err = No such file or directory
Cannot connect to server request channel
jack server is not running or cannot be started
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for 4294967295, skipping unlock
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for 4294967295, skipping unlock
ALSA lib pulse.c:243:(pulse_connect) PulseAudio: Unable to connect: Connection refused

AL lib: (EE) ALCplaybackAlsa_open: Could not open playback device 'default': Connection refused
2018-04-18 02:48:03: ERROR[Main]: An unhandled exception occurred: Audio: Global Initialization: Device Open
2018-04-18 02:48:03: ERROR[Main]: In thread 7ff0ec445900:
2018-04-18 02:48:03: ERROR[Main]: /home/james/minetest/src/main.cpp:e2: int main(int, char**): A fatal error occured: Audio: Global Initialization: Device Open
Aborted
```
For Windows: https://forum.minetest.net/viewtopic.php?f=42&t=17439&p=340937#p340937

To reproduce, one way is to preoccupy the audio device with an ALSA client, so pulseaudio, alsa dmix, or jack or any other sound system; or openal itself cannot open/access the device.

EDIT by SmallJoker: Merged text with #8107

**Comment by 12Me21** [comment]
I'm having a similar problem on windows:

```
2018-09-06 10:04:20: ERROR[Main]: An unhandled exception occurred: Audio: Global Initialization: Device Open
2018-09-06 10:04:20: ERROR[Main]: In thread 1:
2018-09-06 10:04:20: ERROR[Main]: /builds/minetest/minetest/build/minetest/src/main.cpp:e2: main: A fatal error occured: Audio: Global Initialization: Device Open
```

**Comment by jastevenson303** [comment]
Thanks for your assistance, SmallJoker.

I wanted to add, I realize there's no `--nosound`, but there is a CMake option to compile without; but more importantly: There's a `enable_sound = false` in the minetest.conf.

**Comment by GoatSupremecist** [comment]
I run a minetest server for the elementary schools in my school district as a reward time for students. Many of our Windows 10 desktop computers have sound cards that disable the audio device when no headphones/speakers are plugged in. I've tried to write a launcher script to detect if the sound card is disabled and then edit the minetest.conf to set `enable_sound = false` but I've no luck detecting that the sound device is disabled. I would really appreciate it if someone could have a look at this problem and resolve it. Please consider making this bug fix a priority. Myself and thousands of students thank you!
https://forum.minetest.net/viewtopic.php?f=42&t=17439&p=340937#p340937

**Comment by nerzhul** [comment]
sorry i missed some time to fix it... trying to take time to fix it this weekend if possible

**Comment by ashbaught** [comment]
Awesome! We are all very appreciative! Thanks!

**Comment by nerzhul** [comment]
i don't know if it can be possible to do a quick fix or if it needs a major refactor. In the first case it will be fixed for release, in second not :(

**Comment by jastevenson303** [comment]
cant "just" create and open a dummy openal device?

**Comment by ashbaught** [comment]
I was able to circumvent this issue by launching minetest from a script that prompts the user if they want sound or not. Then, the script copies the appropriate minetes.conf from a file share. Its still not an ideal solution because it is  a little confusing for kindergartners that can't read so well, but it is working for the most part.

**Comment by charliek0031** [comment]
> I was able to circumvent this issue by launching minetest from a script that prompts the user if they want sound or not. Then, the script copies the appropriate minetes.conf from a file share. Its still not an ideal solution because it is a little confusing for kindergartners that can't read so well, but it is working for the most part.

And what is this script?

**Comment by Azl-Eyekay** [comment]
this bug is in 5.1.1 too
![Capture](https://user-images.githubusercontent.com/53867511/74502439-85717680-4f13-11ea-9ae9-37f68d5f7d63.PNG)

**Comment by kaincenteno** [comment]
this bug is still present on the minetest ubuntu snap when running on wayland.
```
kain@kain-Blade-Stealth:~$ minetest
2020-05-02 23:31:48: WARNING[Main]: system-wide share not found at "/share/minetest"
2020-05-02 23:31:48: WARNING[Main]: system-wide share found at "/snap/minetest/1521/bin/../share/minetest"
ALSA lib conf.c:3916:(snd_config_update_r) Cannot access file /usr/share/alsa/alsa.conf
ALSA lib pcm.c:2495:(snd_pcm_open_noupdate) Unknown PCM default
AL lib: (EE) ALCplaybackAlsa_open: Could not open playback device 'default': No such file or directory
2020-05-02 23:31:48: ERROR[Main]: An unhandled exception occurred: Audio: Global Initialization: Device Open
2020-05-02 23:31:48: ERROR[Main]: In thread 7fa3d2bd7900:
2020-05-02 23:31:48: ERROR[Main]: /build/minetest/parts/minetest/src/src/main.cpp:e8: int main(int, char**): A fatal error occurred: Audio: Global Initialization: Device Open
Aborted (core dumped)
```

**Comment by nerzhul** [comment]
i just published #9804, if you can test it, it should fix the issue, as it doesn't say that openal failure is a global failure

**Comment by nerzhul** [comment]
Fixed now, sorry for the delay :)
